packages/function-memoize/index.js

```diff
@@ -0,0 +1,47 @@
+module.exports = memoize;
+
+/*
+  const sumByOne = memoize(function(value) {
+    return value + 1;
+  });
+
+  sumByOne(10);
+  sumByOne(10); -- Cache hit!
+
+  sumByOne(20);
+  sumByOne(20); -- Cache hit!
+
+  // Custom cache
+  var sum = memoize(function(a, b) {
+    return a + b
+  }, function(a, b) {
+    return `${a}-${b}`
+  })
+
+  sum(10, 10)
+  sum(10, 20)
+  sum(10, 20) -- Cache hit!
+*/
+
+function memoize(callback, resolver) {
+  if (resolver !== undefined && typeof resolver !== 'function') {
+    throw new Error('`resolver` should be a function');
+  }
+
+  var cache = {};
+
+  var memoized = function() {
+    var args = arguments;
+    var key = resolver ? resolver.apply(this, args) : args[0];
```

Comment:
Can we improve on first argument as default key? How about JSON.stringifying argument array for key if no resolver supplied?

Reply:
Liked the idea of using `JSON. stringifying`.